app/services/magic_link_flow.rb

```diff
@@ -0,0 +1,26 @@
+class MagicLinkFlow
+  def self.sign_up(candidate)
+    magic_link_token = MagicLinkToken.new
+    candidate.update(magic_link_token: magic_link_token.encrypted)
+    link = "http://localhost:3000/personal-details/new?token=#{magic_link_token.raw}"
+    AuthenticationMailer.send_sign_up(to: candidate.email_address, link: link).deliver!
```

Comment:
Consider giving the mailer method a noun rather than a verb name, as we call `deliver!` and that does the verb work?